lib/spack/spack/build_systems/autotools.py

```diff
@@ -95,10 +95,15 @@ class AutotoolsPackage(PackageBase):
     #: Options to be passed to autoreconf when using the default implementation
     autoreconf_extra_args = []
 
+    @run_after('autoreconf')
     def _do_patch_config_guess(self):
         """Some packages ship with an older config.guess and need to have
         this updated when installed on a newer architecture."""
 
+        if not self.patch_config_guess or not self.spec.satisfies(
+                'arch=linux-rhel7-ppc64le'
+        ):
+            return True
```

Comment:
there are other places where True/False are returned. Should I change all the "return True" cases to "return" and throw an exception in the "return False" cases?